File: mifosng-android/src/main/java/com/mifos/objects/collectionsheet/LoanAndClientName.java
package com.mifos.objects.collectionsheet;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by Tarun on 17-07-2017.
 */

public class LoanAndClientName implements Parcelable {

    private LoanCollectionSheet loan;

    private String clientName;
    private  int id;

    public LoanAndClientName(LoanCollectionSheet loan, String clientName, int id) {
        this.loan = loan;
        this.clientName = clientName;
        this.id = id;
    }

    public LoanCollectionSheet getLoan() {
        return loan;
    }

    public String getClientName() {
        return clientName;
    }

    public int getId() {
        return id;
    }


    protected LoanAndClientName(Parcel in) {
        loan = (LoanCollectionSheet) in.readValue(LoanCollectionSheet.class.getClassLoader());
        clientName = in.readString();
        id = in.readInt();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(loan);
        dest.writeString(clientName);
        dest.writeInt(id);
    }

    @SuppressWarnings("unused")
    public static final Parcelable.Creator<LoanAndClientName> CREATOR = new
            Parcelable.Creator<LoanAndClientName>() {
        @Override
        public LoanAndClientName createFromParcel(Parcel in) {
            return new LoanAndClientName(in);
        }

        @Override
        public LoanAndClientName[] newArray(int size) {
            return new LoanAndClientName[size];
        }
    };
}

File: mifosng-android/src/main/java/com/mifos/objects/collectionsheet/LoanAndClientName.kt
package com.mifos.objects.collectionsheet

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import com.mifos.objects.collectionsheet.LoanCollectionSheet

/**
 * Created by Tarun on 17-07-2017.
 */
class LoanAndClientName(
    val loan: LoanCollectionSheet?,
    val clientName: String?,
    val id: Int
) : Parcelable {

    constructor(parcel: Parcel) : this(
        parcel.readParcelable(LoanCollectionSheet::class.java.classLoader),
        parcel.readString(),
        parcel.readInt()
    )

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeParcelable(loan, flags)
        dest.writeString(clientName)
        dest.writeInt(id)
    }

    override fun toString(): String {
        return "LoanAndClientName(loan=$loan, clientName='$clientName', id=$id)"
    }

    companion object CREATOR : Parcelable.Creator<LoanAndClientName> {
        override fun createFromParcel(parcel: Parcel): LoanAndClientName {
            return LoanAndClientName(parcel)
        }

        override fun newArray(size: Int): Array<LoanAndClientName?> {
            return arrayOfNulls(size)
        }
    }
}